### PR TITLE
Integrate SwiftLint and fix CI

### LINKED ISSUE #15: Code style and lint

### PR DESCRIPTION
**Pull request by eyvallah**
This PR does

1. Add lint rules taken from Outlook (with removal of some project specific settings)
2. Integrate SwiftLint with CI and enable "treat warnings as errors" in Swift/Objective-C code in CI (Lint warning will not throw error)

Closes #15

### CODE REVIEW

File: Sources/FluentDarkModeKitExample/ViewController.swift [hunk 1]
```diff
@@ -17,8 +17,7 @@ class ViewController: UIViewController {
     )
   }
 
-  @objc
-  private func refresh() {
+  @objc private func refresh() {
```
Comment (imWildCat) [inline Sources/FluentDarkModeKitExample/ViewController.swift:20]: @kylinroc what do you think of this change?

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -9,9 +9,12 @@ jobs:
...
+        run: sudo xcode-select -s /Applications/Xcode_11.4.1.app
+
+      - name: Install SwiftLint
+        run: brew install swiftlint
 
       - uses: actions/checkout@v1
 
       - name: Build
-        run: xcodebuild -scheme DarkModeKit -destination generic/platform=iOS -configuration Release | xcpretty -s
+        run: xcodebuild -scheme FluentDarkModeKit -destination generic/platform=iOS -configuration Release GCC_TREAT_WARNINGS_AS_ERRORS="YES" SWIFT_TREAT_WARNINGS_AS_ERRORS="YES" | xcpretty -s && exit ${PIPESTATUS[0]}
```
Comment (icodesign) [inline .github/workflows/main.yml:20]: We should enable it in the project.

Comment (icodesign) [inline .github/workflows/main.yml:20]: Just curious: Why we need the `exit` at the end?

File: Sources/FluentDarkModeKitExample/ViewController.swift (hunk 1 above)
Comment (icodesign) [reply Sources/FluentDarkModeKitExample/ViewController.swift:20]: I personally have no strong preference.  But I like the new one.

File: .github/workflows/main.yml (hunk 2 above)
Comment (eyvallah) [reply .github/workflows/main.yml:20]: See usage section of https://github.com/xcpretty/xcpretty

CI needs exit code to determine if the command has succeeded or not, xcpretty eats the exit code from xcodebuild and returns success even if xcodebuild failed. If you check a bit in the CI logs in previous commits (with wrong target `DarkModeKit ` specified), they all failed silently 👀

File: FluentDarkModeKit.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -753,6 +753,7 @@
 				DYLIB_COMPATIBILITY_VERSION = 1;
 				DYLIB_CURRENT_VERSION = 1;
 				DYLIB_INSTALL_NAME_BASE = "@rpath";
+				GCC_TREAT_WARNINGS_AS_ERRORS = YES;
```
Comment (icodesign) [inline FluentDarkModeKit.xcodeproj/project.pbxproj:756]: We don't need this for Debug as this may be annoying when developing.

Comment (eyvallah) [reply FluentDarkModeKit.xcodeproj/project.pbxproj:756]: makes sense, will limit to release only

File: Sources/FluentDarkModeKitExample/ViewController.swift (hunk 1 above)
Comment (kylinroc) [reply Sources/FluentDarkModeKitExample/ViewController.swift:20]: I'm ok with both style.

**Pull request merged**